terraform_files/vsphere/main.tf

```diff
@@ -100,11 +100,15 @@ resource "vsphere_virtual_machine" "master" {
     network_id = data.vsphere_network.network.id
   }
 
-  disk {
-    label            = "master-${count.index}-disk-0"
-    size             = var.master_disk_size_gib
-    eagerly_scrub    = false
-    thin_provisioned = true
+  dynamic "disk" {
+    for_each = range(var.master_disk_count)
+    content {
+      label            = "master-${count.index}-disk-${count.index + 1 + disk.key}"
```

Comment:
what's the idea of ``disk-${count.index + 1 + disk.key}``? why not say ``disk-${count.index + 1}-${disk.key}``?
also couldn't understand what's ``disk.key`` and where it's coming from

Reply:
> what's disk.key and where it's coming from

https://registry.terraform.io/providers/hashicorp/vsphere/latest/docs/resources/virtual_machine#key

Reply:
> why not say disk-${count.index + 1}-${disk.key}

I'll change it, that sounds more reasonable

Reply:
OK, so the label for `disk-${count.index + 1}-${disk.key}` will be `master-0-disk-1-1` for disk number 2 while on `disk-${count.index + 1 + disk.key}` the label will be `master-0-disk-2`.

Reply:
yeah, I understand what's the difference between the options, but I think the addition of the numbers doesn't make sense and might lead to duplicated values (which is solved by having them separately)